hawkular-wildfly-monitor/src/main/java/org/hawkular/agent/monitor/storage/MetricsOnlyMetricDataPayloadBuilder.java

```diff
@@ -33,18 +35,38 @@
 public class MetricsOnlyMetricDataPayloadBuilder implements MetricDataPayloadBuilder {
 
     // key is metric ID, value is list of data points where a data point is a map with timestamp and value
-    private Map<String, List<Map<String, Number>>> allMetrics = new HashMap<>();
+    private Map<String, List<Map<String, Number>>> allGauges = new HashMap<>();
+    private Map<String, List<Map<String, Number>>> allCounters = new HashMap<>();
 
     // a running count of the number of data points that have been added
     private int count = 0;
 
     @Override
-    public void addDataPoint(String key, long timestamp, double value) {
-        List<Map<String, Number>> data = allMetrics.get(key);
+    public void addDataPoint(String key, long timestamp, double value, MetricType metricType) {
+        Map<String, List<Map<String, Number>>> map;
+
+        switch (metricType) {
+            case GAUGE: {
+                map = allGauges;
+                break;
+            }
+            case COUNTER: {
+                map = allGauges; // SHOULD BE allCounters!!!
```

Comment:
when https://issues.jboss.org/browse/HWKMETRICS-166 is fixed, we can change this line to be "map = allCounters;"